hashicorp/packer
# Backport of website: fix husky command  into stable-website

## Backport

This PR is auto-generated from #12821 to be assessed for backporting due to the inclusion of the label backport/website.



The below text is copied from the body of the original PR.

---

### Description

In [another PR](https://github.com/hashicorp/packer/pull/12809) the husky pre-commit hook was changed to match the v9 specs, but `website` being a sub-directory of the root directory containing `.git` was not accounted for. This PR changes the pre-commit hook and script to account for `website` being a sub-directory of the directory containg `.git`, and ensures the pre-commit hook runs as expected. ([Husky doc here](https://typicode.github.io/husky/how-to.html#project-not-in-git-root-directory)).

This PR also adds back in the `dart-linkcheck` package needed for the `linkcheck` script.

### Testing

1. cd into `website/`
2. Run `npm install`
3. Make changes to a file in `website/`
4. Run `git commit -m "<your commit message>"`
5. You should see something like this in your terminal:
```
➜ git commit -m "Fixed typos in website readme"          
✔ Preparing...
✔ Running tasks...
✔ Applying modifications...
✔ Cleaning up...
[<branch name> c06b34955] Fixed typos in website readme
 1 file changed, 2 insertions(+), 2 deletions(-)
 ```

---

<details>
<summary> Overview of commits </summary>

  - 10725dec1848523abb593b585c82deb1594fdff1  - f4e70e15db53ad8fcbff1b1173d3262e5e324b84 

</details>
